src/main/java/io/confluent/copycat/hdfs/partitioner/Partitioner.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright 2015 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ **/
+
+package io.confluent.copycat.hdfs.partitioner;
+
+import org.apache.kafka.copycat.sink.SinkRecord;
+
+import java.util.Map;
+
+/**
+ * Partition incoming records, and generates directories and file names in which to store the
+ * incoming records.
+ */
+public interface Partitioner {
+  void configure(Map<String, Object> config);
+  String encodePartition(SinkRecord sinkRecord);
```

Comment:
Can we add some documentation here? This is actually flexible enough and a nice extension point that people may want to implement their own Partitioners to plug in, especially as they would otherwise have to wait for updated releases. A small amount of guidance could go a long way here.